clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/main/MainIngestionMonitorService.java

```diff
@@ -271,6 +293,10 @@ int resumeOngoingIngestionTasks() {
               LOGGER
                   .info("Recovered ingestion task in isolated process for topic: {}, partition: {}", topic, partition);
               count.addAndGet(1);
+              if (isTopicPartitionInLeaderState(topic, partition)) {
+                client.promoteToLeader(topic, partition);
```

Comment:
I am not sure if there could be a race here. e.g. after L296  but before L297 is executed. Helix decide to demote this host to a standby, and the whole demotion procedure (demoteToStandby) is done before L297, then executing L297 is not proper. If this might happen, looks like VeniceConcurrentHashMap is not enough and we might need a dedicate lock for protection?